<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
    
      <xacro:property name="initial_positions_file" value="$(find arm_moveit)/config/initial_positions.yaml"/>
      <xacro:property name="initial_positions" value="${xacro.load_yaml(initial_positions_file)['initial_positions']}"/>

        <xacro:arg name="use_simulation" default="false" />
      

        <ros2_control name="arm" type="system">
            <hardware>
              <xacro:if value="${use_simulation}">
                <!-- By default, set up controllers for simulation. This won't work on real hardware -->
                <plugin>mock_components/GenericSystem</plugin>
              </xacro:if>
              <xacro:unless value="${use_simulation}">
                <plugin>arm_teensy_hardware/ArmTeensySystem</plugin>
                <param name="port">/dev/ttyACM0</param>
                <param name="baudrate">115200</param>
              </xacro:unless>
            </hardware>
            <joint name="arm_base_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_base_joint']}</param>
                </state_interface>
            </joint>
            <joint name="arm_shoulder_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_shoulder_joint']}</param>
                </state_interface>
            </joint>
            <joint name="arm_elbow_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_elbow_joint']}</param>
                </state_interface>
            </joint>
            <joint name="arm_wrist_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_wrist_joint']}</param>
                </state_interface>
            </joint>
            <joint name="arm_gripper_base_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_gripper_base_joint']}</param>
                </state_interface>
            </joint>
            <joint name="arm_gripper_left_joint">
                <command_interface name="position"/>
                <state_interface name="position">
                  <param name="initial_value">${initial_positions['arm_gripper_left_joint']}</param>
                </state_interface>
            </joint>

        </ros2_control>
    
</robot>
